src/components/input/input.ts

```diff
@@ -299,7 +299,10 @@ export default class SlInput extends LitElement {
     const hasHelpTextSlot = this.hasSlotController.test('help-text');
     const hasLabel = this.label ? true : !!hasLabelSlot;
     const hasHelpText = this.helpText ? true : !!hasHelpTextSlot;
-    const hasClearIcon = this.clearable && !this.disabled && !this.readonly && this.value.length > 0;
+    const hasClearIcon =
+      this.clearable && !this.disabled && !this.readonly && (typeof this.value === 'number' || this.value.length > 0);
+    const isFirefox = typeof window.InstallTrigger !== 'undefined';
```

Comment:
I also thought this will work but in Chrome you will lose your date picker.
To reproduce:
1. Open the Codepen of @oliversalzburg: https://codepen.io/oliversalzburg/pen/vYdMWOe?editors=1000
2. Assign `clip-path: inset(0 2em 0 0);` to the inner input element and see how it disappears